feat: add cli guides (#82)
sapphiredev/website: docs/Guide/CLI/getting-started.md

---
title: Getting Started
sidebar_position: 1
---

import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem';

# Getting Started

## Installation

<Tabs groupId="packageManagers">
<TabItem value="yarn" label="Yarn (Recommended)" default>

```shell:no-line-numbers
$ yarn global add @sapphire/cli
```

</TabItem>

<TabItem value="npm" label="NPM">

```shell:no-line-numbers
$ npm install @sapphire/cli -g
```

</TabItem>

> </Tabs>

Reviewer: ![image](https://user-images.githubusercontent.com/4019718/139582037-91774bdd-28f4-4dc4-a55a-9cb5c5b2154b.png)

codeblocks don't let me post it as code

Author: ?

Author: @Favna evaluate

more about it here.

```shell:no-line-numbers{6-8}
$ sapphire new
✔ What's the name of your project? … my-sapphire-bot
✔ Choose a language for your project › TypeScript (Recommended)
✔ Choose a template for your project › Default template (Recommended)
✔ What format do you want your config file to be in? › YAML
? What package manager do you want to use? › - Use arrow-keys. Return to submit.
❯   Yarn (Recommended)
    npm
```

Choose a package manager for your project, we will choose **Yarn** in this guide.

```shell:no-line-numbers{7}
$ sapphire new
✔ What's the name of your project? … my-sapphire-bot
✔ Choose a language for your project › TypeScript (Recommended)
✔ Choose a template for your project › Default template (Recommended)
✔ What format do you want your config file to be in? › YAML
✔ What package manager do you want to use? › Yarn (Recommended)
? Do you want to create a git repository for this project? › (y/N)
```

Lastly, if you want to initialize a git repository in your project, type `y` and enter. If you don't want a git
repository in your project, just enter.

Now wait for the CLI to create your project and install the dependencies. This should take ~2 minutes.

> ```shell:no-line-numbers

Reviewer: No need for `:no-line-numbers`. That's a vuepress thing anyway, it doesn't do anything for Docusaurus. Line numbers aren't enabled for shell.

Also make it `bash` instead of `shell`, that follows the rest of thee shell blocks.

Also for 
```
```shell:no-line-numbers{7}
```
change that to 
```
```bash {7}
```
so note that space.

Reviewer: This has not been done for all code blocks @UndiedGamer

Reviewer: Just to be clear, I want to see absolute 0 usages of triple backticks "shell". It doesn't get syntax highlighting, whereas bash does.

$ sapphire new
✔ What's the name of your project? … my-sapphire-bot
✔ Choose a language for your project › TypeScript (Recommended)
✔ Choose a template for your project › Default template (Recommended)
✔ What format do you want your config file to be in? › YAML
✔ What package manager do you want to use? › Yarn (Recommended)
✔ Do you want to create a git repository for this project? … no
✔ Cloning the repository
✔ Setting up the project
✔ Installing dependencies using Yarn
Done!
```

Congratulations, you created a bot using Sapphire CLI. Read the README to see what you need to do before running it.

> :::: code-group ::: code-group-item Bash

Reviewer: Change all of these to the `<Tabs>` structure as seen in other pages. Docusaurus doesn't do grouping like this.

---
title: Getting Started
sidebar_position: 1
---

import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem';

## Installation

```bash npm2yarn2pnpm
npm install @sapphire/cli -g
```

> :::caution

Reviewer 2: ```suggestion
:::info
```
Feels like it should be more of a pointer than a warning to me.

> You can install the CLI as a devDependency, but we recommend installing it globally.

Reviewer 2: ```suggestion
You can install the CLI as a development dependency, but we recommend installing it globally so you can reuse the utility in multiple projects.
```

:::

<details>

> <summary>New project</summary>

Reviewer 2: ```suggestion
<summary>Setting up a new project</summary>
```

## Creating a bot

```bash
sapphire new
```

> Run this command to start.

Reviewer 2: This is roughly the equivalent of top-posting in emails. Please put instructions *before* the demonstrations to be consistent and to preserve the logical flow of a guide.

If it helps, try to remember how research is done: method first, practical application second, comments and further reading last.

Author: @enxg

Reviewer: Done

✔ Choose a template for your project › Default template (Recommended)
✔ What format do you want your config file to be in? › YAML
✔ What package manager do you want to use? › Yarn (Recommended)
? Do you want to create a git repository for this project? › (y/N)
```

Lastly, if you want to initialize a git repository in your project, type `y` and enter. If you don't want a git
repository in your project, just enter.

Now wait for the CLI to create your project and install the dependencies. This should take ~2 minutes.

```bash
$ sapphire new
✔ What's the name of your project? … my-sapphire-bot
✔ Choose a language for your project › TypeScript (Recommended)
✔ Choose a template for your project › Default template (Recommended)
✔ What format do you want your config file to be in? › YAML
✔ What package manager do you want to use? › Yarn (Recommended)
✔ Do you want to create a git repository for this project? … no
✔ Cloning the repository
✔ Setting up the project
✔ Installing dependencies using Yarn
Done!
```

Congratulations, you created a bot using Sapphire CLI. Read the README to see what you need to do before running it.

<Tabs>

> <TabItem value="bash" label="Bash" default>

Reviewer 2: This is by no means bash exclusive - preferably refer to it as UNIX shell, or just shell, or anything of that nature so long it isn't a specific exclusive shell.

Author: @Favna does docusaurus support posix as a code block language?

```bash
$ cat my-sapphire-bot/README.md
# TypeScript Sapphire Bot example

This is a basic setup of a Discord bot using the [sapphire framework][sapphire] written in TypeScript
...
```

</TabItem>

<TabItem value="ps1" label="Powershell">

```bash
$ Get-Content .\my-sapphire-bot\README.md
# TypeScript Sapphire Bot example

This is a basic setup of a Discord bot using the [sapphire framework][sapphire] written in TypeScript
...
```

</TabItem>

<TabItem value="commandprompt" label="CMD">

```bash
$ type my-sapphire-bot\README.md
# TypeScript Sapphire Bot example

This is a basic setup of a Discord bot using the [sapphire framework][sapphire] written in TypeScript
...
```

</TabItem>
</Tabs>

</details>

<details>
<summary> Existing project </summary>

## Using Sapphire CLI on an existing project

> ```shell

Reviewer: ```suggestion
```bash
```
I said no more `shell`. Don't make me teach you about "Find in Files" ಠ_ಠ

shell is not a docusaurus supported language: https://github.com/FormidableLabs/prism-react-renderer/blob/master/src/vendor/prism/includeLangs.js

$ sapphire init
```

Run this command to start.

```shell {2-4}
$ sapphire init
? What format do you want your config file to be in? › - Use arrow-keys. Return to submit.
    JSON
❯   YAML
```

Choose a format for your config file, we will choose **YAML** in this guide. This config file is for Sapphire CLI, read
more about it here.

> ```shell {3-5}

Reviewer: ```suggestion
```bash {3-5}
```

$ sapphire init
✔ What format do you want your config file to be in? › YAML
? Choose the language used in your project › - Use arrow-keys. Return to submit.
❯   TypeScript
    JavaScript
```

Choose the language you are using in your project, we are using **TypeScript** in this one, so we will go with it.

> ```shell {4}

Reviewer: ```suggestion
```bash {4}
```

> $ sapphire init

Reviewer: ```suggestion
sapphire init
```

✔ What format do you want your config file to be in? › YAML
✔ Choose the language used in your project › TypeScript
? Your base directory › src
```

Enter your base directory, where your project's entry point is located, defaults to `src`.

> ```shell {5}

Reviewer: ```suggestion
```bash {5}
```

> $ sapphire init

Reviewer: ```suggestion
sapphire init
```

✔ What format do you want your config file to be in? › YAML
✔ Choose the language used in your project › TypeScript
✔ Your base directory … src
? Where do you store your commands? (do not include the base) › commands
```

Enter your commands directory, where your commands are located (don't include the base), defaults to `commands`.

> ```shell {6}

Reviewer: ```suggestion
```bash {6}
```

> $ sapphire init

Reviewer: ```suggestion
sapphire init
```

✔ What format do you want your config file to be in? › YAML
✔ Choose the language used in your project › TypeScript
✔ Your base directory … src
✔ Where do you store your commands? (do not include the base) … commands
? Where do you store your listeners? (do not include the base) › listeners
```

Enter your listeners directory, where your listeners are located (don't include the base), defaults to `listeners`.

> ```shell {7}

Reviewer: ```suggestion
```bash {7}
```

> $ sapphire init

Reviewer: ```suggestion
sapphire init
```

✔ What format do you want your config file to be in? › YAML
✔ Choose the language used in your project › TypeScript
✔ Your base directory … src
✔ Where do you store your commands? (do not include the base) … commands
✔ Where do you store your listeners? (do not include the base) … listeners
? Where do you store your arguments? (do not include the base) › arguments
```

Enter your arguments directory, where your arguments are located (don't include the base), defaults to `arguments`.

> ```shell {8}

Reviewer: ```suggestion
```bash {8}
```

> $ sapphire init

Reviewer: ```suggestion
sapphire init
```

✔ What format do you want your config file to be in? › YAML
✔ Choose the language used in your project › TypeScript
✔ Your base directory … src
✔ Where do you store your commands? (do not include the base) … commands
✔ Where do you store your listeners? (do not include the base) … listeners
✔ Where do you store your arguments? (do not include the base) … arguments
? Where do you store your preconditions? (do not include the base) › preconditions
```

Enter your preconditions directory, where your preconditions are located (don't include the base), defaults to
`preconditions`.

> ```shell {9}

Reviewer: ```suggestion
```bash {9}
```

> $ sapphire init

Reviewer: ```suggestion
sapphire init
```

✔ What format do you want your config file to be in? › YAML
✔ Choose the language used in your project › TypeScript
✔ Your base directory … src
✔ Where do you store your commands? (do not include the base) … commands
✔ Where do you store your listeners? (do not include the base) … listeners
✔ Where do you store your arguments? (do not include the base) … arguments
✔ Where do you store your preconditions? (do not include the base) … preconditions
? Do you want to enable custom file templates? › (y/N)
```

Choose if you want to enable custom file (piece) templates, type `y` and enter if you want to enable it. If you don't
want to enable it, just enter.

> ```shell

Reviewer: ```suggestion
```bash
```

> $ sapphire init

Reviewer: ```suggestion
sapphire init
```

✔ What format do you want your config file to be in? › YAML
✔ What package manager do you want to use? › Yarn (Recommended)
✔ Do you want to create a git repository for this project? … no
✔ Cloning the repository
✔ Setting up the project
✔ Installing dependencies using Yarn
Done!
```

Congratulations, you created a bot using Sapphire CLI. Read the README to see what you need to do before running it.

<Tabs>
<TabItem value="bash" label="Bash" default>

```bash
$ cat my-sapphire-bot/README.md
# TypeScript Sapphire Bot example

This is a basic setup of a Discord bot using the [sapphire framework][sapphire] written in TypeScript
...
```

</TabItem>

<TabItem value="ps1" label="Powershell">

```bash
$ Get-Content .\my-sapphire-bot\README.md
# TypeScript Sapphire Bot example

This is a basic setup of a Discord bot using the [sapphire framework][sapphire] written in TypeScript
...
```

</TabItem>

<TabItem value="commandprompt" label="CMD">

```bash
$ type my-sapphire-bot\README.md
# TypeScript Sapphire Bot example

This is a basic setup of a Discord bot using the [sapphire framework][sapphire] written in TypeScript
...
```

</TabItem>
</Tabs>

</details>

<details>

> <summary> Existing project </summary>
> # Using Sapphire CLI on an existing project

Reviewer 3: fix this
![](https://reds.is-inside.me/Tu626xPi.png)
```suggestion
<summary> Existing project </summary>

## Using Sapphire CLI on an existing project
```
